ruma-api-macros/src/api.rs

```diff
@@ -227,7 +227,7 @@ impl ToTokens for Api {
 
         let request_lifetimes = self.request.combine_lifetimes();
 
-        let non_auth_endpoint_impls = if requires_authentication.value {
+        let non_auth_endpoint_impls = if authentication.to_string() != "None" {
```

Comment:
You shouldn't need `to_string`, AFAIK Idents can be compared to strong literals directly. We may also want to convert this into an enum when going from RawApi > Api and match against its variants here.